feat: Adds ath9k-htc wifi driver support

acbrazzl opened the pull request:
Adds the following packages to the build:
- kmod-ath9k-htc
- ath9k-htc-firmware
- batctl-default

Removes the following packages from the build:
- wpad-basic-mbedtls
- batctl-full

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

google-labs-jules[bot] committed:
feat: Update OpenWrt package selection

Adds the following packages to the build:
- kmod-ath9k-htc
- ath9k-htc-firmware
- batctl-default

Removes the following packages from the build:
- wpad-basic-mbedtls
- batctl-full

The following requested packages were already enabled:
- kmod-batman-adv
- luci-proto-batman-adv
- wpad-openssl

coreywagehoft commented:
@acbrazzl could you provide more details about the requested changes?

acbrazzl commented:
> @acbrazzl could you provide more details about the requested changes?

Hey @coreywagehoft, I had gathered up a list of packages to include on a build to get ath9k-based wifi modules working on OpenWRT with the intent of enabling dual & tri-band buildouts w/ Halow as the backbone.  So then I had intended to eventually commit this to the OpenManet build tool chain ...
... and was trying out Google's Jules agentic AI ... and tasked it to add these packages to the OpenManet build :)

So aside from Jules claiming this builds, I haven't done an end-to-end test yet.  I'll throw this into draft and report back when it's proved out.  Stylistically, it may just be worth noting the preferred method of OpenWRT package injection in the README?

coreywagehoft commented:
> > @acbrazzl could you provide more details about the requested changes?
> 
> Hey @coreywagehoft, I had gathered up a list of packages to include on a build to get ath9k-based wifi modules working on OpenWRT with the intent of enabling dual & tri-band buildouts w/ Halow as the backbone. So then I had intended to eventually commit this to the OpenManet build tool chain ... ... and was trying out Google's Jules agentic AI ... and tasked it to add these packages to the OpenManet build :)
> 
> So aside from Jules claiming this builds, I haven't done an end-to-end test yet. I'll throw this into draft and report back when it's proved out. Stylistically, it may just be worth noting the preferred method of OpenWRT package injection in the README?

Thank you for the detail @acbrazzl.  I have been testing dual and tri-band setups.  I have been working with both Intel cards and several using ath10k.  I will say I have had nothing but issues with them.  I have been testing several cards using the ath10k drivers.

Now I may be testing a different setup than you are thinking.  I am working on having both 900Mhz HaLow and 2.4ghz WiFi bonded in batman-adv.  This will allow batman to select the best link and route over it.  The assumption I am working with is then 5ghz would be used for EUD devices to connect to the radio.  This is where I run into issues with both intel and ath* drivers.  

It would not allow the card to run both in 802.11s mode for mesh and AP mode for client devices.  You need a Dual Band Dual Concurrent (DBDC) card to do this.  DBDC cards show up as 2 separate radios in OpenWRT allowing you to configure them independently for different bands.

I am currently testing some Mediatek based cards and I am having more success with this, but I am not ready to release it until I finish further testing.

I am not against adding drivers for different wifi cards for different use cases, but they need to be well tested before merging.

I do however request you remove the changes for `batctl` and `wpad-basic-mbedtls`.  The size on disk difference between `batctl-full` and `batctl-default` is only a few KB.  I would prefer to keep the full setup for debugging purposes.  The `wpad-basic-mbedtls` comes from the upstream Morse Micro configuration, and I don't want to introduce any potential side effects with how morse micro interfaces with wpad.

google-labs-jules[bot] committed:
feat: Add initial support for Raspberry Pi 5 (BCM2712)

This change adds the basic scaffolding for building an OpenWrt image for the Raspberry Pi 5.

It includes:
- A new `bcm2712` subtarget for the `bcm27xx` target.
- A new `ekh-bcm2712` board configuration.
- A `README.md` file in the new board directory detailing the progress and the remaining steps.

The work is currently blocked by an issue with creating symlinks in the development environment.
